ruby/ql/lib/codeql/ruby/frameworks/Files.qll

```diff
@@ -110,72 +110,81 @@ module IO {
    *
    * This class includes reads both from shell commands and reads from the
    * filesystem. For working with filesystem accesses specifically, see
-   * `IOFileReader` or the `FileSystemReadAccess` concept.
+   * `FileReader` or the `FileSystemReadAccess` concept.
```

Comment:
This doesn't include `File` calls anymore though, by the looks of it?

Reply:
What I mean is, this comment implies that `FileReader` is a subset of `IOReader` but it seems that they're disjoint. Is that right?

Reply:
I should have updated this comment - it's pretty misleading. 

`IOReader` only includes calls that use the `IO` class directly - so not any calls involving subclasses like `File`.

`FileReader` includes calls using both `File` and `IO` - but it excludes calls using `IO` that "obviously" invoke shell commands such as `IO.read("|date")` where its argument is a `StringlikeLiteral` that specifies a shell command. It's not a very precise exclusion, as a shell command could read from a file anyway, or, conversely, a non-`StringlikeLiteral` argument could refer to something that isn't a filename (e.g. `cmd = "|date"; IO.read(cmd)`).

I've updated the doc comments on both of these classes to try to make this clearer, relying more on examples than on prose.